### PR TITLE
Fix/python gil release on rdkit threadsafe sss only

### PR DESCRIPTION
**Pull request by bp-kelley**
The GIL is only released when RDK_BUILD_THREADSAFE_SSS is on.

Additional python treading tests were added (RDK_TEST_MULTITHREADED).  These only check for sanity errors (i.e. re-entering the python interpreter when the GIL is released) not the underlying C++ code.

### CODE REVIEW

File: Code/GraphMol/Wrap/testThreads.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    core_smarts = '[#6]-!@[#6]-!@[#8]-!@[#6]:1:[#6](-!@[#6]#!@[#7]):[#6](-!@[#7]):[#6]:[#6](-!@[#7]-!@[#6](-!@[#6]-!@[#6]:2:[#6]:[#6]:[#6]:[#6]:[#6]:2)=!@[#8]):[#7]:1'
+    ref_mol = Chem.MolFromMolBlock(ref_sdf)
+    if ref_mol is None:
+        raise ValueError('Bad ref structure')
+    core_mol = Chem.MolFromSmarts(core_smarts)
+    if core_mol is None:
+        raise ValueError('Bad core structure')
+    ref_mol.GetSubstructMatch(core_mol)
+
+def sub_match2():
```
Comment (greglandrum) [inline Code/GraphMol/Wrap/testThreads.py]: This is really repetitive; couldn't you just have one test function that takes the method to call as an argument?

File: Code/GraphMol/Wrap/testThreads.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    core_smarts = '[#6]-!@[#6]-!@[#8]-!@[#6]:1:[#6](-!@[#6]#!@[#7]):[#6](-!@[#7]):[#6]:[#6](-!@[#7]-!@[#6](-!@[#6]-!@[#6]:2:[#6]:[#6]:[#6]:[#6]:[#6]:2)=!@[#8]):[#7]:1'
+    ref_mol = Chem.MolFromMolBlock(ref_sdf)
+    mol = Chem.MolToBinary(ref_mol)
+    
+nthreads = 100
+for i in xrange(1, nthreads):
```
Comment (greglandrum) [inline Code/GraphMol/Wrap/testThreads.py]: xrange doesn't work on python3

**Comment by bp-kelley** [comment]
Should we turn on thread safety for testing?

**Comment by bp-kelley** [comment]
You got xrange but missed the print statement ;)

**Comment by greglandrum** [comment]
damn!

**Pull request merged**